Fix remaining flake8 violations

cottsay opened the pull request:
Follow-up to #608

dirk-thomas commented:
> W504 line break after binary operator
The code base follows W503 (Line break occurred before a binary operator) and therefore W504 needs to be ignored.

> I202 Additional newline in a group of imports

Since this patch already addresses one of these cases how about addressing them all?

Also if we want to avoid regressions the CI build should run the linter.

cottsay commented:
> ... W504 needs to be ignored.

Done.

> ... this patch already addresses one of these cases how about addressing them all?

The line I added earlier wasn't related to `I202`, but I went ahead and fixed all of the `I202` anyway.

> ... the CI build should run the linter.

I added it to the beginning of the matrix - it runs quicker than the other jobs, and is likely to break most often. I also bumped it to Python 3.6 - for some reason, Python 3.4 had a load of debug statements turned on and it exceeded the log size.

dirk-thomas commented on .travis.yml:
```diff
@@ -15,6 +11,14 @@ env:
+      sudo: false
+      install:
+        - pip install catkin-pkg EmPy PyYAML rosdistro vcstool
+        - pip install coverage "flake8<3.7" flake8-docstrings flake8-import-order nose pep8 pyflakes
```
Why `flake8 < 3.7`? Also `coverage` doesn't seem to be used.

It would be good to install more `flake8` plugins (like the ones we use in `ament_flake8`). But that can be a separate PR.

cottsay replied:
`flake8` 3.7 includes at least one breaking change to the legacy API, and my machine didn't have 3.7, and the original tests didn't target it. So it was the least amount of effort.

I went ahead and made the corresponding change to `test_flake8.py`.

I can't see any flake8 plugin references in `ament_flake8`. If you want those included in this PR, you'll have to be more specific, but this is already a pretty big batch of goodness without them.

dirk-thomas replied:
Most of the plugins are not available as Debian packages therefore they are not listed in the package manifest. I was referring to the plugins we install in CI: https://github.com/ros2/ci/blob/3eb0842d60d4a88aaaa02a0acbf02485c4f364aa/ros2_batch_job/__main__.py#L53-L61

But as said above: this can be done in a separate PR.

cottsay replied:
> this can be done in a separate PR

Cool, sounds good.

dirk-thomas commented on .travis.yml:
```diff
@@ -15,6 +11,14 @@ env:
+        - pip install catkin-pkg EmPy PyYAML rosdistro vcstool
+        - pip install coverage "flake8<3.7" flake8-docstrings flake8-import-order nose pep8 pyflakes
+      script:
+        - python setup.py nosetests -s --tests test
```
I would lean towards using `pytest` nowadays. But that could also be done in a separate PR.

cottsay replied:
Sure

dirk-thomas commented on test/test_flake8.py:
```diff
@@ -23,7 +23,7 @@ def test_flake8_conformance():
         '--ignore=%s' % ','.join([
             'D100', 'D101', 'D102', 'D103', 'D104', 'D105',
-            'E501']),
+            'E501', 'W504']),
```
Use `--extend-ignore` instead.

cottsay replied:
This is only available in flake8 3.7, but since I updated to use that, it should work.

dirk-thomas commented on setup.py:
```diff
@@ -32,6 +32,11 @@
         'empy',
         'PyYAML'],
+    'tests_require': [
+        'flake8 > 3.7',
```
`>=`?
